python/cuml/dask/extended/linear_model/logistic_regression.py

```diff
@@ -0,0 +1,210 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from cuml.dask.common.base import BaseEstimator
+from cuml.common import with_cupy_rmm
+from cuml.common.import_utils import has_daskglm
+
+import cupy as cp
+import numpy as np
+from dask.utils import is_dataframe_like, is_series_like, is_arraylike
+import cudf
+
+
+class LogisticRegression(BaseEstimator):
+    """
+    Distributed Logistic Regression for Binary classification.
+
+
+    Parameters
+    ----------
+    fit_intercept: boolean (default = True)
+       If True, the model tries to correct for the global mean of y.
+       If False, the model expects that you have centered the data.
+    solver : 'admm'
+        Solver to use. Only admm is supported currently.
+    penalty : {'l1', 'l2'} (default = 'l2')
+        Regularization technique for the solver.
+    C: float (default = 1.0)
+       Inverse of regularization strength; must be a positive float.
+    max_iter: int (default = 100)
+        Maximum number of iterations taken for the solvers to converge.
+    verbose : int or boolean (default=False)
+        Sets logging level. It must be one of `cuml.common.logger.level_*`.
+        See :ref:`verbosity-levels` for more info.
+
+    Attributes
+    ----------
+    coef_: device array (n_features, 1)
+        The estimated coefficients for the logistic regression model.
+    intercept_: device array (1,)
+        The independent term. If `fit_intercept` is False, will be 0.
+    solver: string
+        Algorithm to use in the optimization process. Currently only `admm` is
+        supported.
+
+    Notes
+    ------
+
+    This estimator is a wrapper class around Dask-GLM's
+    Logistic Regression estimator. Several methods in this wrapper class
+    duplicate code from Dask-GLM to create a user-friendly namespace.
+    """
+
+    def __init__(
+        self,
+        client=None,
+        fit_intercept=True,
+        solver="admm",
+        penalty="l2",
+        C=1.0,
+        max_iter=100,
+        verbose=False,
+        **kwargs
+    ):
+        super(LogisticRegression, self).__init__(
+            client=client, verbose=verbose, **kwargs
+        )
+
+        if not has_daskglm("0.2.1.dev"):
+            raise ImportError(
+                "dask-glm >= 0.2.1.dev was not found, please install it"
+                " to use multi-GPU logistic regression."
+            )
+
+        from dask_glm.estimators import LogisticRegression \
+            as LogisticRegressionGLM
+
+        self.fit_intercept = fit_intercept
+        self.solver = solver
+        self.penalty = penalty
```

Comment:
Need to add check so error is raised for None/elasticnet

Reply:
Resolved by https://github.com/rapidsai/cuml/pull/3512/commits/5493497f0efec14ff49cf539627c2ccd958b8c83